src/coreclr/nativeaot/Runtime/thread.cpp

```diff
@@ -769,6 +801,51 @@ bool Thread::InternalHijack(PAL_LIMITED_CONTEXT * pSuspendCtx, void * pvHijackTa
     return fSuccess;
 }
 
+#ifdef FEATURE_SUSPEND_REDIRECTION
+CONTEXT* Thread::GetRedirectionContext()
+{
+    if (m_redirectionContext == NULL)
+    {
+        m_redirectionContext = PalAllocateCompleteOSContext(&m_redirectionContextBuffer);
+    }
+
+    return m_redirectionContext;
+}
+
+bool Thread::Redirect(PAL_LIMITED_CONTEXT * pSuspendCtx)
+{
+    if (IsDoNotTriggerGcSet())
+        return false;
+
+    CONTEXT* redirectionContext = GetRedirectionContext();
+    if (redirectionContext == NULL)
+        return false;
+
+    StackFrameIterator frameIterator(this, pSuspendCtx);
+
+    if (!frameIterator.IsValid())
+        return false;
+
+    frameIterator.CalculateCurrentMethodState();
```

Comment:
Why call `CalculateCurrentMethodState`? `frameIterator` is not used after this point.

Reply:
It was copied from `InternalHijack` initially. I now think the entire `frameIterator` deal is unnecessary here.